### ISSUE TITLE
Warning about usage of undeclared variable

### ISSUE THREAD

**Issue by jeanmonod** [kalessil/phpinspectionsea#1771]
Here is a situation where I would have been very happy to get a warning or an error message. Here is a method

    public function getFirstUser()
    {
        $users = $this->getList();
        if (count($users)>0) {
            return reset($user);
        }
    }

Did you see the mistake? It's the `reset($user)` instead of `reset($users)`. I haven't seen it and EA either. Would't been worth it to highlight this issue? At least with a warning.

BTW, it's the same behaviour with `array_unshift`

Thanks in advance for the answer

**Comment by kalex375** [comment]
Default PHPStorm inspector show error:
![image](https://user-images.githubusercontent.com/966329/134520299-3b30d8b7-c2a5-4911-bb4d-e1e725e624af.png)

**Comment by jeanmonod** [comment]
@kalex375 yes, it show an error if you do a `return $user`, but not with a `return reset($user)`. I guess it's because reset or array_unshift accept a variable passed by reference. So it's can technically valid. But from my pov it make no sense...

**Comment by jdreesen** [comment]
This is how PhpStorm warns me (I'm using PhpStorm 2021.2.1):

![grafik](https://user-images.githubusercontent.com/424602/134590425-18a72d5c-7b30-4853-a389-144e7e5b5f62.png)

**Comment by jeanmonod** [comment]
@jdreesen That's interesting, because on my side, I have nothing:

![image](https://user-images.githubusercontent.com/114489/134614598-1dc6ea26-af7f-49ec-b9ae-bcc6382dcd36.png)

I'm under IntelliJ IDEA 2020.3.3 (Ultimate Edition)
Build #IU-203.7717.56, built on March 15, 2021
PHP Language level: 8.0

And as far as I know, without custom settings. Can I have something wrongly configured? Any idea why?

**Comment by ea-inspections-team** [comment]
Side-note: the analyzer inspections are prefixed with '[EA] '. In this case it was a core inspection.
